### DISCUSSION TITLE
disable or enable components，UI controls can change accordingly

### DISCUSSION THREAD

**Discussion by bopresslte** [langflow-ai/langflow#1191] [Ideas]
hi, I have two ideas
1. Is there a plan to provide functionality to disable or enable components, so that certain system provided components can be retained or disabled according to preferences
2. The controls within the component are now independent of each other. Will it be possible to add components that can be associated with responses? For example, if the value of a dropdown changed, the selection options of other dropdown will also change accordingly

**Comment by ogabrielluiz** [comment]
Hey, @bopresslte 

We have a prototype for something similar in the dev branch.
Essentially, you pass a callable (any function) to the "options" key and that will be called whenever you click a refresh button on the Component in the UI. 

What you are proposing is something we have been thinking for a long time but haven't quite figured out how due to time constraints. 

One idea I had the other day was a Portal field. You'd set a component name and a field name, and the value on the component with that name and that field would dictate the value on the component you defined this Portal field.


The solution to your idea would have a similar implementation to what the Refresh has: when the field changes in the frontend, the value of that field would be passed to a predefined callable that would do something based on the value. 

Now, changing things in the template from within a method would require some more consideration.

Have you had any thoughts on the implementation?

Thanks for the idea!

**Reply by bopresslte** [reply]
Are you referring to backend definition and frontend rendering?

My idea is similar https://github.com/alibaba/formily ，use json config.

```json
"x-reactions": {
  "dependencies": [
	{
	  "property": "value",
	  "type": "string",
	  "source": "sa2f7fuc1n5", // frontend component id
	  "name": "v_z4sezsk03f1"
	}
  ],
  "fulfill": {
	"state": {
	  "visible": "{{$deps.v_z4sezsk03f1=1}}" // current componet behavior
	}
  }
}
```

Everything happens in the front-end.

can define config in `build_config` of custom component class 

This may require a lot of development time, I think this is a good interactive experience and worth waiting for, Thanks

**Comment by Cristhianzl** [comment]
Hi! 👋

We are using other channels as our official means of communication with users. We apologize for the delayed response.
For faster assistance, please direct your questions to our Discord channel or open a GitHub issue, as we are much more active on these platforms.

[Langflow Discord Server](https://discord.com/invite/EqksyE2EX9)

Thank you for your understanding.

Best regards,
The Langflow Team
